apps-script/contact-form-app/rolodex.gs

```diff
@@ -0,0 +1,341 @@
+    return null;}
+    }
+
+/**
+* Opens a dialog that prompts users to add details about a contact.
+*
+*
+* @return {Object} open a dialog.
+*/
+function openDialog(event) {
+  return {
+    "actionResponse": {
+      "type": "DIALOG",
+      "dialogAction": {
+        "dialog": {
+          "body": {
+            "sections": [
+              {
+                "header": "Add new contact",
+                "widgets": contactFormWidgets.concat([{
+                  "buttonList": {
+                    "buttons": [{
+                      "text": "Review and submit",
+                        "onClick": {
+                          "action": {
+                            "function": "openNextCard"
+                            }
+                          }
+                        }
+                      ]
+                    }
+                  }
+                ])
+              }
+            ]
+          }
+        }
+      }
+    }
+  };
+}
+
+/**
+* Returns another dialog or card message that displays a confirmation of
+* contact details before users submit.
+*
+* @param {String} the user who submitted the information.
+* @param {String} contactName the contact name from the previous dialog or card.
+* @param {String} contactBirthdate the birthdate from the previous dialog or card.
+* @param {String} contactType the contact type from the previous dialog or card.
+* @param {boolean} whether the information was submitted from a dialog.
+*
+* @return {Object} returns a dialog or private card message.
+*/
+function openNextCard(user, contactName, contactBirthdate, contactType, fromDialog) {
+  const name = contactName ?? "<i>Not provided</i>";
+  const birthdate = contactBirthdate ?? "<i>Not provided</i>";
+  const type = contactType ?? "<i>Not provided</i>";
+  const cardConfirmation =
+              {
+                "header": "Your contact",
+                "widgets": [
+                  {"textParagraph": {"text": "Confirm contact information and submit:"}},
+                  {"textParagraph": {"text": "<b>Name:</b> " + name}},
+                  {"textParagraph": {"text": "<b>Birthday:</b> " + convertMillisToDateString(birthdate)}},
+                  {"textParagraph": {"text": "<b>Type:</b> " + type}},
+                  {"buttonList": {
+                      "buttons": [
+                        {"text": "Submit",
+                          "onClick": {
+                            "action": {
+                              "function": "submitForm",
+                               "parameters": [
+                                 {"key": "contactName",
+                                  "value": name
+                                 },
+                                 {"key": "contactBirthdate",
+                                  "value": birthdate
+                                 },
+                                 {"key": "contactType",
+                                  "value": type
+                    }]}}}]}}]};
```

Comment:
The indentation doesn't seem correct here.

Reply:
Reopening because it seems this change is not applied. When the object contains multiple properties, the opening bracket must be alone in the line, and each property on its own line.

Reply:
Updated, but feel free to suggest an edit if I didn't address fully. I don't get any formatitng errors when I run this, and I'm trying to reduce the lines of brackets in the code per Pierrick's preference.